server/src/test/java/org/apache/uniffle/server/ShuffleServerTest.java

```diff
@@ -79,4 +80,53 @@ public void startTest() {
     }
 
   }
+
+  @ParameterizedTest
+  @ValueSource(booleans = {true, false})
+  public void decommissionTest(boolean shutdown) throws Exception {
+    ShuffleServerConf serverConf = createShuffleServerConf();
+    serverConf.set(SERVER_DECOMMISSION_CHECK_INTERVAL, 1000L);
+    serverConf.set(SERVER_DECOMMISSION_SHUTDOWN, shutdown);
+    ShuffleServer shuffleServer = new ShuffleServer(serverConf);
+    shuffleServer.start();
+    assertEquals(ServerStatus.ACTIVE, shuffleServer.getServerStatus());
+    // Shuffle server is not decommissioning, but we can also cancel it.
+    shuffleServer.cancelDecommission();
+    ShuffleTaskManager shuffleTaskManager = shuffleServer.getShuffleTaskManager();
+    String appId = "decommissionTest_appId_" + shutdown;
+    shuffleTaskManager.registerShuffle(appId, 0, Lists.newArrayList(), new RemoteStorageInfo("/tmp"), "");
+    shuffleServer.decommission();
+    assertEquals(ServerStatus.DECOMMISSIONING, shuffleServer.getServerStatus());
+    // Shuffle server is decommissioning, but we can also decommission it again.
+    shuffleServer.decommission();
+    shuffleServer.cancelDecommission();
+    shuffleTaskManager.removeResources(appId);
+    // Wait for 2 seconds, make sure cancel command is work.
+    Thread.sleep(2000);
+    assertEquals(ServerStatus.ACTIVE, shuffleServer.getServerStatus());
+    shuffleServer.decommission();
+    if (shutdown) {
+      Awaitility.await().timeout(10, TimeUnit.SECONDS).until(
+          () -> !shuffleServer.isRunning());
+    } else {
+      Awaitility.await().timeout(10, TimeUnit.SECONDS).until(
+          () -> ServerStatus.DECOMMISSIONED.equals(shuffleServer.getServerStatus()));
+      assertEquals(true, shuffleServer.isRunning());
+      shuffleServer.stopServer();
+    }
+  }
+
+  private ShuffleServerConf createShuffleServerConf() throws Exception {
+    ShuffleServerConf serverConf = new ShuffleServerConf();
+    serverConf.setInteger(ShuffleServerConf.RPC_SERVER_PORT, 9527);
+    serverConf.setString(ShuffleServerConf.RSS_STORAGE_TYPE, StorageType.LOCALFILE.name());
+    serverConf.setBoolean(ShuffleServerConf.RSS_TEST_MODE_ENABLE, true);
+    serverConf.setInteger(ShuffleServerConf.JETTY_HTTP_PORT, 9528);
+    serverConf.setString(ShuffleServerConf.RSS_COORDINATOR_QUORUM, "localhost:0");
+    serverConf.set(ShuffleServerConf.RSS_STORAGE_BASE_PATH, Arrays.asList("/tmp/null"));
```

Comment:
Is `"/tmp/null"` OK here?
Better use JUnit managed `@TempDir` here.

Reply:
`createShuffleServerConf` is referenced by multiple unit test cases, we need add `@TempDir` to every cases? Is there a better way?

Reply:
```
--- a/server/src/test/java/org/apache/uniffle/server/ShuffleServerTest.java
+++ b/server/src/test/java/org/apache/uniffle/server/ShuffleServerTest.java
@@ -17,12 +17,14 @@

 package org.apache.uniffle.server;

+import java.io.File;
 import java.util.Arrays;
 import java.util.concurrent.TimeUnit;

 import com.google.common.collect.Lists;
 import org.awaitility.Awaitility;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.ValueSource;

@@ -39,6 +41,9 @@ import static org.junit.jupiter.api.Assertions.fail;

 public class ShuffleServerTest {

+  @TempDir
+  private File tempDir;
+
   @Test
   public void startTest() {
     try {
@@ -123,7 +128,7 @@ public class ShuffleServerTest {
     serverConf.setBoolean(ShuffleServerConf.RSS_TEST_MODE_ENABLE, true);
     serverConf.setInteger(ShuffleServerConf.JETTY_HTTP_PORT, 9528);
     serverConf.setString(ShuffleServerConf.RSS_COORDINATOR_QUORUM, "localhost:0");
-    serverConf.set(ShuffleServerConf.RSS_STORAGE_BASE_PATH, Arrays.asList("/tmp/null"));
+    serverConf.set(ShuffleServerConf.RSS_STORAGE_BASE_PATH, Arrays.asList(tempDir.getAbsolutePath()));
     serverConf.setLong(ShuffleServerConf.DISK_CAPACITY, 1024L * 1024L * 1024L);
     serverConf.setLong(ShuffleServerConf.SERVER_BUFFER_CAPACITY, 100);
     serverConf.setLong(ShuffleServerConf.SERVER_READ_BUFFER_CAPACITY, 10);
```

Reply:
Get it, thanks.